src/components/drawer/drawer.tsx

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import classNames from 'classnames';
+import styles from './drawer.module.scss';
+
+interface DrawerProps {
+    open: boolean;
+    onClose: () => void;
+    children: React.ReactNode;
+}
+
+export const Drawer = ({ open, onClose, children }: DrawerProps) => {
```

Comment:
Very basic version without animation (original website doens't have it too), trapping focus and blocking the scrollbar.